3836 terraform provider (#3914)
astronomer/docs: astro/terraform-provider.md

---
title: 'Astro Terraform Provider'
sidebar_label: 'Astro Terraform Provider'
id: terraform-provider
description: Learn about the Astro Terraform Provider and how to work with it.
---

> [Terraform by HashiCorp](https://www.terraform.io/) allows you to manage your infrastructure as code. This means you can programmatically manage resources, user access, and networking using Terraform and Terraform Providers.

Author: Alternative to other ways of creating Deployments

Terraform providers act as an interface between Terraform and different platforms or services that you might use in your infrastructure implementation. You can use the [Astro Terraform Provider](https://github.com/astronomer/terraform-provider-astro) to programmatically manage your Astro resources, Deployments, and user management.

The Astro Terraform Provider is open source and available through the [Terraform registry](https://registry.terraform.io/providers/astronomer/astro/0.3.0). You can also clone the provider from its [GitHub Repo](https://github.com/astronomer/terraform-provider-astro), if you also want to adapt it for your custom use case, or submit a suggesiont with a pull request.

> 

Author: Add that details about provider commands are documented in the registry + on github and that only examples of use on Astro are here.

---
title: 'Astro Terraform Provider'
sidebar_label: 'Astro Terraform Provider'
id: terraform-provider
description: Learn about the Astro Terraform Provider and how to work with it.
---

:::publicpreview
:::

> [Terraform by HashiCorp](https://www.terraform.io/) allows you to manage your infrastructure as code. This means you can programmatically manage resources, user access, and networking using Terraform and Terraform Providers. This provides you with an alternative way to use the Astro CLI, Astro UI, or Astro API to create Deployments and manage users.

Reviewer: ```suggestion
[Terraform by HashiCorp](https://www.terraform.io/) allows you to manage your infrastructure as code. This means you can programmatically manage resources, user access, and networking using Terraform and Terraform Providers. This provides you with an alternative to the Astro CLI, Astro UI, or Astro API to manage resources.
```

> [Terraform by HashiCorp](https://www.terraform.io/) allows you to manage your infrastructure as code. This means you can programmatically manage resources, user access, and networking using Terraform and Terraform Providers. This provides you with an alternatve way to using the Astro CLI, Astro UI, or Astro API to create Deployments and manage users.

Reviewer 2: ```suggestion
[Terraform by HashiCorp](https://www.terraform.io/) allows you to manage your infrastructure as code. This means you can programmatically manage resources, user access, and networking using Terraform and Terraform Providers. This provides you with an alternative way to use the Astro CLI, Astro UI, or Astro API to create Deployments and manage users.
```

> Terraform providers act as an interface between Terraform and different platforms or services that you might use in your infrastructure implementation. You can use the [Astro Terraform Provider](https://github.com/astronomer/terraform-provider-astro) to programmatically manage your Astro resources, Deployments, and users.

Reviewer: Terraform can do more than just create Deployment, and manage users - perhaps we can make these ending statements a bit more general and just say manage your astro resources (eg. Workspaces, Deployents, users, etc)

> Terraform providers act as an interface between Terraform and different platforms or services that you might use in your infrastructure implementation. You can use the [Astro Terraform Provider](https://github.com/astronomer/terraform-provider-astro) to programmatically manage your Astro resources, Deployments, and users.

Author: ```suggestion
Terraform providers act as an interface between Terraform and different platforms or services that you might use in your infrastructure implementation. You can use the [Astro Terraform Provider](https://github.com/astronomer/terraform-provider-astro) to programmatically manage your Astro resources.
```

> Terraform providers act as an interface between Terraform and different platforms or services that you might use in your infrastructure implementation. You can use the [Astro Terraform Provider](https://github.com/astronomer/terraform-provider-astro) to programmatically manage your Astro resources, Deployments, and user management.

Reviewer 2: ```suggestion
Terraform providers act as an interface between Terraform and different platforms or services that you might use in your infrastructure implementation. You can use the [Astro Terraform Provider](https://github.com/astronomer/terraform-provider-astro) to programmatically manage your Astro resources, Deployments, and users.
```

> The Astro Terraform Provider is open source and available through the [Terraform registry](https://registry.terraform.io/providers/astronomer/astro/0.3.0). You can also clone the provider from its [GitHub Repo](https://github.com/astronomer/terraform-provider-astro), if you also want to adapt it for your custom use case, or submit a suggesiont with a pull request.

Reviewer 2: ```suggestion
The Astro Terraform Provider is open source and available through the [Terraform registry](https://registry.terraform.io/providers/astronomer/astro/0.3.0). You can also clone the provider from its [GitHub Repo](https://github.com/astronomer/terraform-provider-astro) if you also want to adapt it for your custom use case, or submit a suggestion with a pull request.
```
